catalog/security/catalog-security-ingestplugin/src/main/java/ddf/catalog/security/ingest/IngestPlugin.java

```diff
@@ -0,0 +1,137 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.security.ingest;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.subject.Subject;
+
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+
+/**
+ * IngestPlugin is a PreIngestPlugin that restricts the create/update/delete operations
+ * on the catalog to a group defined by a set of configurable user attributes.
+ */
+public class IngestPlugin implements PreIngestPlugin {
+
+    private String permissionsString;
+
+    private List<KeyValuePermission> permissions = new ArrayList<>();
+
+    public CreateRequest process(CreateRequest createRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.CREATE_ACTION);
+        return createRequest;
+    }
+
+    public UpdateRequest process(UpdateRequest updateRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.UPDATE_ACTION);
+        return updateRequest;
+    }
+
+    public DeleteRequest process(DeleteRequest deleteRequest)
+            throws PluginExecutionException, StopProcessingException {
+        authenticateUser(CollectionPermission.DELETE_ACTION);
+        return deleteRequest;
+    }
+
+    /**
+     * Getter used by the framework to populate the configuration ui
+     *
+     * @return
+     */
+    public String getPermissionsString() {
+        return this.permissionsString;
+    }
+
+    /**
+     * Get the KeyValuePermission that have been parsed from the permissions string
+     *
+     * @return
+     */
+    public List<KeyValuePermission> getPermissions() {
+        return Collections.unmodifiableList(permissions);
+    }
+
+    /**
+     * Setter used by the ui to set the permissions/attributes
+     *
+     * @param permString
+     */
+    public void setPermissionsString(String permString) {
+        this.permissionsString = permString;
+        parsePermissionsFromString(this.permissionsString);
+    }
+
+    /**
+     * Authenticates the current Subject against the configured ingest permissions.
+     * If the subject doesn't have the right credentials than a StopProcessingException
+     * will be thrown
+     *
+     * @throws StopProcessingException
+     */
+    private void authenticateUser(String action) throws StopProcessingException {
```

Comment:
This looks generic enough to move to security app.  It should also return a boolean instead of controlling flow with an exception.